### PR TITLE
Add MAM Resources

### LINKED ISSUE #3490: [Bug] MSAL.NET does not pass intune_mam_resource to the broker
**Logs and network traces**
NA

**Which version of MSAL.NET are you using?**
4.45.0

**Platform**
iOS

**What authentication flow has the issue?**
* Desktop / Mobile
    * [X] Interactive
    * [ ] Integrated Windows Authentication
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web app 
    * [ ] Authorization code
    * [ ] On-Behalf-Of
* Daemon app 
    * [ ] Service to Service calls

Other?
NA

**Is this a new or existing app?**
New

**Expected behavior**
intune_mam_resource  should be passed to the broker. Empty if there is not entry.

**Actual behavior**
intune_mam_resource  should be passed to the broker. Empty if there is not entry.

**Possible solution**
Add the entry to parameters

**Additional context / logs / screenshots / links to code**
NA

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
Fixes #3490 
Adds intune mam resource to request

**Changes proposed in this request**
Adding unconditional MAM resource to the request

**Testing**
Manual with Intune.

**Performance impact**
None

**Documentation**
None

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs [hunk 1]
```diff
@@ -129,11 +129,12 @@ private void ValidateRedirectUri(Uri redirectUri)
...
             brokerRequest.Add(BrokerParameter.ClientVersion, MsalIdHelper.GetMsalVersion());
+
             var realEnrollmentId = IntuneEnrollmentIdHelper.GetRawEnrollmentId();
-            if (!string.IsNullOrEmpty(realEnrollmentId))
-            {
-                brokerRequest.Add(BrokerParameter.IntuneEnrollmentIds, realEnrollmentId);
-            }
+            brokerRequest.Add(BrokerParameter.IntuneEnrollmentIds, realEnrollmentId ?? string.Empty);
+
+            var intuneMAMResource = IntuneEnrollmentIdHelper.GetRawMAMResources();
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs:136]: nit: `Mam`, not `MAM`

File: src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs [hunk 2]
```diff
@@ -52,6 +53,16 @@ internal static string GetRawEnrollmentId()
 #endif
         }
 
+        internal static string GetRawMAMResources()
+        {
+#if iOS
+            var keychainData = NSUserDefaults.StandardUserDefaults.StringForKey(Intune_MAMResourceKey);
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs:59]: Does IntuneSDK put this resource into the KeyChain? What does it look like, is it just normal resource URI that we're trying to access?

File: src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs [hunk 3]
```diff
@@ -14,6 +14,7 @@ namespace Microsoft.Identity.Client.Platforms.iOS
     internal class IntuneEnrollmentIdHelper
     {
         const string EnrollmentIdKey = "intune_app_protection_enrollment_id_V1";
+        const string Intune_MAMResourceKey = "intune_mam_resource_V 1";
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs:17]: Is the space here intended?

File: src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs [hunk 4]
```diff
@@ -129,11 +129,12 @@ private void ValidateRedirectUri(Uri redirectUri)
             brokerRequest.Add(BrokerParameter.ClientId, authenticationRequestParameters.AppConfig.ClientId);
             brokerRequest.Add(BrokerParameter.CorrelationId, authenticationRequestParameters.RequestContext.CorrelationId.ToString());
             brokerRequest.Add(BrokerParameter.ClientVersion, MsalIdHelper.GetMsalVersion());
+
             var realEnrollmentId = IntuneEnrollmentIdHelper.GetRawEnrollmentId();
-            if (!string.IsNullOrEmpty(realEnrollmentId))
-            {
-                brokerRequest.Add(BrokerParameter.IntuneEnrollmentIds, realEnrollmentId);
-            }
+            brokerRequest.Add(BrokerParameter.IntuneEnrollmentIds, realEnrollmentId ?? string.Empty);
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs:134]: In Intune scenario, in which step of the flow would these properties be empty? Is it on the very first request when the device is unregistered? So broker will return an exception?
I assume in non-Intune scenario broker will just ignore these properties?

**Review by pmaytak — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs [hunk 5]
```diff
@@ -52,6 +53,16 @@ internal static string GetRawEnrollmentId()
 #endif
         }
 
+        internal static string GetRawMAMResources()
+        {
+#if iOS
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs:58]: This file should only for iOS, is there a need for the #if ?

File: src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs (hunk 4 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs:134]: This is a great question and the answer should be added as a comment in the code.

**Review by bgavrilMS — APPROVED** [review]
Approved with comments. The "V 1" instead of "V1" seems blocking.

File: src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs (hunk 2 above)
Comment (SameerK-MSFT) [reply src/client/Microsoft.Identity.Client/Platforms/iOS/IntuneEnrollmentIdHelper.cs:59]: > Does IntuneSDK put this resource into the KeyChain? 
Yes
> What does it look like?
A json string.

**Review by trwalke — APPROVED** [review]

**Pull request merged**